### PR TITLE
Finalise adding type hints for arelle.plugin.validate.esef

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
This enforces strict typing for the entire `arelle.plugin.validate.esef` package.

#### Description of change
This PR:
- Cleans up white space in a number of files
- Removes unused imports
- Adds type hints in `arelle.plugin.validate.ESEF.__init__` and files used herein.
- Moves type hints for `_` (gettext, I believe) to a common file.

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/ModelDtsObject.py
Comment (ggravlingen) [inline arelle/ModelDtsObject.py:602]: There's a lot of noise here but this is what's actually changed.

File: arelle/PackageManager.py
Comment (ggravlingen) [inline arelle/PackageManager.py:378]: Lots of noise here too, but main changes where made here.

File: arelle/plugin/validate/ESEF/common.py
Comment (ggravlingen) [inline arelle/plugin/validate/ESEF/common.py:5]: We can probably move this even higher up in the hierarchy but I've placed it here for now.

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [inline arelle/plugin/validate/ESEF/__init__.py:104]: My typing skills are not yet strong enough to handle functions that yield. I hope to become better so that I can fix this.

Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py:104]: Looks good, thanks!

Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py:104]: Oh wait, it took a while for VS code to process after I changed it. Getting 94 new errors now.
```
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "str" of "Optional[str]" has no attribute "contextID"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "contextID"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "str" of "Optional[str]" has no attribute "value"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "value"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:830: error: Item "str" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:830: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:864: error: Item "str" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:864: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:881: error: Item "_Element" of "Union[ModelObject, _Element]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:881: error: "_Element" has no attribute "qname"  [attr-defined]
arelle/plugin/validate/ESEF/__init__.py:919: error: Assignment to variable "err" outside except: block  [misc]
arelle/plugin/validate/ESEF/__init__.py:1049: error: Item "str" of "Optional[str]" has no attribute "concept"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1049: error: Item "None" of "Optional[str]" has no attribute "concept"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1053: error: Item "str" of "Optional[str]" has no attribute "unit"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1053: error: Item "None" of "Optional[str]" has no attribute "unit"  [union-attr]
```

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py]: Just changed it. There are two more instances of `assert modelXbrl.modelDocument is not None`. If we expect `modelDocument` not to be `None`, could we simply return if it's `None`?

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py]: I'm not sure I want to go down this rabbit hole for the moment. It's probably a lot easier to fix when we've added more type hints throughout the code.

File: arelle/plugin/validate/ESEF/__init__.py
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/__init__.py:113]: Python typing has special handling for `*args` and `**kwargs`.

`(*args: Any, **kwargs: Any)`
is equivalent to
`(*args: tuple[Any, ...], **kwargs: dict[str, Any])`

File: arelle/ModelXbrl.py
Comment (ggravlingen) [reply arelle/ModelXbrl.py:416]: `Any` was actually intentional. `_roleType.genLabel` and `_roleType.definition` both return `Any`, while `roleUriTitle` is `str`. Maybe it's better to keep with just `Any` for now until `genLabel` and `definition` have been typed?

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py]: Looks like these are present in the other plugins as well so I created shared types that we can reuse for the other files.

File: arelle/ModelXbrl.py
Comment (ggravlingen) [reply arelle/ModelXbrl.py:416]: I have given this some thought and I guess definition and genlabel are both str so maybe we should just cast here?

File: arelle/plugin/validate/ESEF/__init__.py
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/__init__.py:417]: Having two very similar variable names is error prone and likely to introduce bugs (you missed renaming some of the instances of `elt` below).

Revert the variable renames and cast here instead:
```python
for uncast_elt, depth in etreeIterWithDepth(ixdsHtmlRootElt):
    # Elsewhere in this module, we can use isinstance(elt, _Element) for typing purposes
    # If we do that here, tests fail. For now we cast elt as Any.
    elt = cast(Any, uncast_elt)
```

File: arelle/ModelXbrl.py
Comment (austinmatherne-wk) [reply arelle/ModelXbrl.py:416]: Yes, it looks like both `_roleType.genLabel` and `_roleType.definition` are `str | None` while the fallback to `self.roleUriTitle` is `str` so this should always return `str`.

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [reply arelle/plugin/validate/ESEF/__init__.py:417]: Nice!

**Comment by ggravlingen** [comment]
Please allow me to rebase this once all suggested changes have been approved.

File: arelle/plugin/validate/ESEF/__init__.py
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/__init__.py:114]:
```suggestion
def disclosureSystemConfigURL(disclosureSystem: DisclosureSystem, *args: Any, **kwargs: Any) -> str:
```

File: arelle/plugin/validate/ESEF/__init__.py
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/__init__.py:180]: `Typing.Dict` [is deprecated](https://docs.python.org/3/library/typing.html#typing.Dict). Use `dict` instead:
```suggestion
    authorityValidations = cast(dict[Any, Any], authorityValidations)
```

**Review by austinmatherne-wk — COMMENTED** [review]
Very close. Small couple of tweaks, but otherwise looks good. 👍

File: arelle/ModelDocument.py
Comment (sagesmith-wf) [inline arelle/ModelDocument.py:530]: Can we get typehints for the params as well? Here and other places

**Comment by ggravlingen** [comment]
> Very close. Small couple of tweaks, ~but otherwise looks good~ (esef conformance tests [are failing](https://github.com/Arelle/Arelle/runs/7828341686?check_suite_focus=true)).

I had some issues with failing tests then filtering on `_Element`. I expect this run to fail and if it does, I'll implement the uncast_solution in all instances in `__init__` to see if that solves it.

**Review by sagesmith-wf — APPROVED** [review]
It might be nice to PR the whitespace fixes separately and just do a sweep over the project with a formatter one time. Some of the fixes seem unrelated to the the changes you were making for this PR (ex. some files don't even add types but are PR'd with whitespace fixes)

**Comment by ggravlingen** [comment]
> It might be nice to PR the whitespace fixes separately and just do a sweep over the project with a formatter one time. Some of the fixes seem unrelated to the the changes you were making for this PR (ex. some files don't even add types but are PR'd with whitespace fixes)

I have thought about this as well. I could just open all 200 something files in VS code, hit save, commit and make a PR. It would make it a lot easier to continue the work.

File: arelle/ModelDocument.py
Comment (ggravlingen) [reply arelle/ModelDocument.py:530]: Generally, I haven't touched type hints for params in ancillary files where I'm not sure what they are. I'm expecting we'll get there in due time.

**Review by austinmatherne-wk — DISMISSED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Pull request merged**
